ui/src/pages/clearing/AllocationModal.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useEffect, useState } from 'react';
+import { useLedger } from '@daml/react';
+import { CreateEvent } from '@daml/ledger';
+import {
+  AssetDeposit,
+  AssetDeposit_SetObservers,
+  AssetDeposit_Split,
+} from '@daml.js/da-marketplace/lib/DA/Finance/Asset';
+import { Button, Form, Header, Modal } from 'semantic-ui-react';
+
+type MarginCallProps = {
+  deposit: CreateEvent<AssetDeposit> | undefined;
+  title: string;
+  open: boolean;
+  onClose: (open: boolean) => void;
+  observers: AssetDeposit_SetObservers;
+};
+
+const AllocationModal: React.FC<MarginCallProps> = ({
+  deposit,
+  title,
+  open,
+  onClose,
+  observers,
+}) => {
+  const ledger = useLedger();
+  const [allocation, setAllocation] = useState<number>(0.0);
+
+  useEffect(() => {
+    if (!!deposit) setAllocation(parseFloat(deposit.payload.asset.quantity));
+  }, [open, deposit]);
+
+  if (!deposit) return <></>;
+
+  const requestAllocation = async () => {
+    const depositQuantity = parseFloat(deposit.payload.asset.quantity);
+
+    if (allocation <= 0 || allocation > depositQuantity) {
+      onClose(false);
+      return;
```

Comment:
error message here? "The allocated quantity must be greater than or equal to 0 and less than the deposit"

Reply:
I could try and convert this over to `ModalFormErrorHandled` so we display the error message to the user? Would you want this as part of this PR or another one ?